src/rp2_common/pico_atomic/pico_atomic.c

```diff
@@ -192,56 +193,56 @@ bool __atomic_compare_exchange_2(volatile void *mem, void *expected, uint16_t de
     return result;
 }
 
-uint32_t __atomic_fetch_add_4(volatile void *mem, uint32_t val, __unused int model) {
-    volatile uint32_t *ptr = mem;
-    uint32_t state = __atomic_lock(mem);
-    uint32_t result = *ptr;
+unsigned int __atomic_fetch_add_4(volatile void *mem, unsigned int val, __unused int model) {
```

Comment:
It looks like this (and the similiar `foo_4` functions) should probably have remained as
```c
uint32_t __atomic_fetch_add_4(volatile void *mem, uint32_t val, __unused int model) {
```
?

Esepcially as the other functions have signatures like:
```c
uint8_t __atomic_fetch_add_1(volatile void *mem, uint8_t val, __unused int model) {
uint16_t __atomic_fetch_add_2(volatile void *mem, uint16_t val, __unused int model) {
uint64_t __atomic_fetch_add_8(volatile void *mem, uint64_t val, __unused int model) {
```

Reply:
I agree but there is a GCC build here when -Wextra is enabled.  See https://github.com/raspberrypi/pico-sdk/actions/runs/9848449043/job/27190379443. The current linux cmake configuration does not enable -Wextra but I'm not clear how to enable it as my CMake skill are not great.

Reply:
Ah, sorry! I made this comment from just looking at the `diff` view on GitHub, before realising that you'd actually specifically commented on this already!! :facepalm:

Reply:
No, no I was hacking up a comment at the same time as you were reviewing and commenting.